src/pages/single-post.tsx

```diff
@@ -0,0 +1,109 @@
+            posts: [],
+            slug: props.match.params.slug
+        };
+    }
+
+    componentDidMount() {
+        const { slug }: any = this.state;
+        fetch("https://admin.cuhacking.com/ghost/api/v3/content/posts/slug/" + slug + "?key=4ff22bcd0f6effc7d2df8ee82b&include=tags,authors")
+            .then(res => res.json())
+            .then(
+                (result) => {
+                    console.log(result);
+                    this.setState({
+                        isLoaded: true,
+                        posts: result.posts
+                    });
+                },
+                (error) => {
+                    this.setState({
+                        isLoaded: true,
+                        error
+                    });
+                }
+            )
+    }
+
+    render() {
+        const { error, isLoaded, posts }: any = this.state;
+        console.log(posts[0]);
+        if (error) {
+            return <div>Error: {error.message}</div>;
+        } else if (!isLoaded) {
+            return <div>Loading...</div>;
+        } else {
+            return (<main id="site-main" className="site-main outer">
+                <div className="inner">
+                    <article className="post-full post">
+                        <header className="post-full-header">
+
+                            <section className="post-full-tags">
+                                {posts[0].tags.map((tag: any) => (<span className="tag-single-post">{tag.name.toUpperCase()}</span>))}
+                            </section>
+
+                            <h1 className="post-full-title">{posts[0].title}</h1>
+
+                            <p className="post-full-custom-excerpt">{posts[0].excerpt}</p>
+
+                            <div className="post-full-byline">
+
+                                <section className="post-full-byline-content">
+
+                                    <ul className="author-list">
+                                        {posts[0].authors.map((author: any) => (
+                                            <li className="author-list-item">
+                                                <div className="author-card">
+                                                    <img className="author-profile-image" src={author.profile_image} alt="Ghost" />
+                                                    <div className="author-info">
+                                                        <div className="bio">
+                                                            <h2>{author.name}</h2>
+                                                        </div>
+                                                    </div>
+                                                </div>
+                                                <div className="author-avatar">
+                                                    <img className="author-profile-image" src="https://static.ghost.org/v3.0.0/images/ghost.png" alt="Ghost" />
```

Comment:
This avatar should be the one that belongs to the author, not the default ghost one. Clicking on it should also lead to the author page, `/blog/author/<author_slug>`.